integrations/ollama/src/haystack_integrations/components/generators/ollama/chat/chat_generator.py

```diff
@@ -102,21 +145,33 @@ def from_dict(cls, data: Dict[str, Any]) -> "OllamaChatGenerator":
         :returns:
             Deserialized component.
         """
+        deserialize_tools_inplace(data["init_parameters"], key="tools")
+
         init_params = data.get("init_parameters", {})
-        serialized_callback_handler = init_params.get("streaming_callback")
-        if serialized_callback_handler:
+
+        if serialized_callback_handler := init_params.get("streaming_callback"):
             data["init_parameters"]["streaming_callback"] = deserialize_callable(serialized_callback_handler)
         return default_from_dict(cls, data)
 
-    def _message_to_dict(self, message: ChatMessage) -> Dict[str, str]:
-        return {"role": message.role.value, "content": message.text}
-
-    def _build_message_from_ollama_response(self, ollama_response: ChatResponse) -> ChatMessage:
+    def _build_message_from_ollama_response(self, ollama_response: "ChatResponse") -> ChatMessage:
         """
         Converts the non-streaming response from the Ollama API to a ChatMessage.
         """
         response_dict = ollama_response.model_dump()
-        message = ChatMessage.from_assistant(response_dict["message"]["content"])
+
```

Comment:
Why not have ollama message -> ChatMessage conversion function as well. It'll be easier to understand code due to symmetry with _convert_message_to_ollama_format and also we'll centralize the conversion logic in one easy to locate, understand and test place

Reply:
I created a conversion function in line with your suggestion: `_convert_ollama_response_to_chatmessage`.
I also made other simplifications. Please take a look.